app/stac_api/models.py

```diff
@@ -125,48 +137,52 @@ class Collection(models.Model):
     # furthermore GeoDjango and its functionality will be used for that.
     # TODO: overwrite items save() function accordingly
     # suggestions of fields to be auto-populated:
-    extent = ArrayField(models.FloatField(), blank=True)  # [Float], auto-populated from items
-
+    extent = models.JSONField(
+        blank=True,
+        null=True,
+        default=get_default_extent_value,
+        encoder=DjangoJSONEncoder,
+        editable=False
+    )
     collection_name = models.CharField(unique=True, max_length=255)  # string
     # collection_name is what is simply only called "id" in here:
     # http://ltboc.infra.bgdi.ch/static/products/data.geo.admin.ch/apitransactional.html#operation/createCollection
     item_type = models.CharField(default="Feature", max_length=20)  # string
-
     keywords = models.ManyToManyField(Keyword)
     license = models.CharField(max_length=30)  # string
     providers = models.ManyToManyField(Provider)
-
     stac_extension = ArrayField(
         models.CharField(max_length=255), default=get_default_stac_extensions, editable=False
     )
     stac_version = models.CharField(max_length=10)  # string
-
     # "summaries" values will be updated on every update of an asset inside the
     # collection
-    summaries_eo_gsd = ArrayField(models.FloatField(), blank=True, null=True)
-    summaries_proj = ArrayField(models.IntegerField(), blank=True, null=True)
-    # after discussion with Chris and Tobias: geoadmin_variant will be an
-    # array field of CharFields. Simple validation is done (e.g. no "Sonderzeichen"
-    # in array)
-    # geoadmin_variant will be also auto-populated on every
-    # update of an asset
-    geoadmin_variant = ArrayField(models.CharField(max_length=15), blank=True, null=True)
-
+    summaries = models.JSONField(
+        blank=True,
+        null=True,
+        default=get_default_summaries_value,
+        encoder=DjangoJSONEncoder,
+        editable=False
+    )
     title = models.CharField(blank=True, max_length=255)  # string
 
     def __str__(self):
         return self.collection_name
 
     def clean(self):
-        # TODO: move this check to the items save()
-        if self.start_date is None and self.end_date is None:
-            raise ValidationError(_('At least a start date or an end date has to be defined.'))
-
         # very simple validation, raises error when geoadmin_variant strings contain special
         # characters or umlaut.
-        for variant in self.geoadmin_variant:
-            if not bool(re.search('^[a-zA-Z0-9]*$', variant)):
-                raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+
+        try:
+            for variant in self.summaries["geoadmin:variant"]:
+                if not bool(re.search('^[a-zA-Z0-9]*$', variant)):
+                    raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
```

Comment:
The error message could also give hints on what was wrong (we don't have security implications here, so we can very well tell the user what exactly was wrong or what we expect)

Reply:
error messages are now more verbose